spec/controllers/api/encryptables_controller_spec.rb

```
@@ -700,6 +701,169 @@
+      expect(duplicated_encryptable.folder_id).not_to eq(file1.folder_id)
+      expect(duplicated_encryptable.encrypted_transfer_password).present?
+      expect(duplicated_encryptable.sender_id).present?
+      expect(duplicated_encryptable.sender_id).to eq(bob.id)
+      expect(duplicated_encryptable.content_type).to eq("text/plain")
+    end
+
+    it 'does not send encryptable credentials to api user' do
+      login_as(:alice)
+
+      share_encryptable_params = {
+        data: {
+          attributes: {
+            id: credentials1.id,
+            receiver_id: api_user.id
+          }
+        }
+      }
+      expect { post :create, params: share_encryptable_params, xhr: true }.to raise_error(StandardError, "Cant transfer to API user")
+    end
+
+    it 'does not send encryptable file to api user' do
+      login_as(:bob)
+
+      share_encryptable_params = {
+        data: {
+          attributes: {
+            id: file1.id,
+            receiver_id: api_user.id
+          }
+        }
+      }
+      expect { post :create, params: share_encryptable_params, xhr: true }.to raise_error(StandardError, "Cant transfer to API user")
+    end
+
+    it 'does not send encryptable credentials to non existing user' do
+      login_as(:alice)
+
+      share_encryptable_params = {
+        data: {
+          attributes: {
+            id: credentials1.id,
+            receiver_id: 5647836758538
+          }
+        }
+      }
+      expect { post :create, params: share_encryptable_params, xhr: true }.to raise_error(StandardError, "Receiver user not found")
+    end
+
+    it 'does not send encryptable file to non existing user' do
+      login_as(:bob)
+
+      share_encryptable_params = {
+        data: {
+          attributes: {
+            id: file1.id,
+            receiver_id: 657656756567856
+          }
+        }
+      }
+      expect { post :create, params: share_encryptable_params, xhr: true }.to raise_error(StandardError, "Receiver user not found")
```

put this with do ... end on multiple lines

---

this should raise user not found exception by active record.
so no need to implement this yourself. check comments further up